docs/source/algorithms/LoadInstrument-v1.rst

```diff
@@ -9,51 +9,82 @@
 Description
 -----------
 
-Loads an instrument definition file (:ref:`IDF <InstrumentDefinitionFile>`)
-into a workspace, which contains information about detector positions,
-their geometric shape, slit properties, links between values stored in
-ISIS log-files and components of the instrument and so on. For more on
-IDFs see: :ref:`InstrumentDefinitionFile <InstrumentDefinitionFile>`.
+Loads the instrument geometry from either an instrument definition file
+(:ref:`IDF <InstrumentDefinitionFile>`) or a Nexus file into a workspace. The
+geometry contains information about detector positions, their geometric shape,
+slit properties, links between values stored in log-files and components of the
+instrument and so on. For more on IDFs see:
+:ref:`InstrumentDefinitionFile <InstrumentDefinitionFile>`.
 
 By default the algorithm will write a 1:1 map between the spectrum
 number and detector ID. Any custom loading algorithm that calls this as
 a Child Algorithm will therefore get this 1:1 map be default. If the
 custom loader is to write its own map then it is advised to set
 ``RewriteSpectraMap`` to false to avoid extra work.
 
-The instrument to load can be specified by either the InstrumentXML,
-Filename and InstrumentName properties (given here in order of
-precedence if more than one is set). At present, if the InstrumentXML is
-used the InstrumentName property should also be set.
+The instrument to load can be specified by either the ``InstrumentXML`` (IDFs
+only), ``Filename`` and ``InstrumentName`` properties (given here in order of
+precedence if more than one is set). At present, if the ``InstrumentXML`` is
+used the ``InstrumentName`` property should also be set.
+
+The ``Filename`` can either be an absolute or relative path. In the latter case,
+both the ``datasearch.directories`` and instrument directories (accessible via
+``getInstrumentDirectories()`` and ``setInstrumentDirectories()``) are searched for
+the specified file. The filename is required to be of the form
+``InstrumentName + _Definition + Identifier + extension``.
+The identifier then is the part of a filename that identifies the instrument
```

Comment:
I think this has already been documented here http://docs.mantidproject.org/nightly/concepts/InstrumentDefinitionFile.html#idf-filename-convention so you might want to link directly to that.

Reply:
Also it might be worth mentioning that there is a difference in behaviour as far as the `InstrumentDataService` is concerned relating to file type. Does this also affect `LoadEmptyInstrument`. If so maybe I forgot to update that, and would appreciate it getting fixed there.

Reply:
What difference are you refering to regarding the `InstrumentDataService`? I'm not sure I follow.

We probably also need to update `LoadEmptyInstrument` to load nexus geometry from instrument name

Reply:
Not so worried about the duplication, more about consistency. Also many algorithm users don't go looking in the wiki, so with the high detail in the wiki on IDFs, it could be a good thing to make users aware of.

Reply:
As I understand the logic at the moment, we do not put instruments loaded via the nexus geometry route in the cache `InstrumentDataService`, but this does happen to instruments loaded via IDF.

Reply:
So do we actually want to add the instrument loaded via nexus to the `InstrumentDataService`? I think it can easily be added

Reply:
The instrument view will work without the instrument service. We should probably add it for consistency.

Reply:
So will we need a lock to make it thread safe, like for the IDF loader
```C++
    // Make InstrumentService access thread-safe
    std::lock_guard<std::recursive_mutex> lock(m_mutex);
```

Reply:
I believe that it would not otherwise be write safe.